machines/backup.ts

```diff
@@ -0,0 +1,220 @@
+                  target: 'failure',
+                },
+                {
+                  target: 'fetchDataFromDB',
+                },
+              ],
+            },
+          },
+          fetchDataFromDB: {
+            entry: ['fetchAllDataFromDB'],
+            on: {
+              STORE_RESPONSE: {
+                actions: 'setDataFromStorage',
+                target: 'writeDataToFile',
+              },
+            },
+          },
+          writeDataToFile: {
+            invoke: {
+              src: 'writeDataToFile',
+            },
+            on: {
+              FILE_NAME: {
+                actions: 'setFileName',
+                target: 'zipBackupFile',
+              },
+            },
+          },
+          zipBackupFile: {
+            invoke: {
+              src: 'zipBackupFile',
+              onDone: {
+                target: 'success',
+              },
+              onError: {
+                target: 'failure',
+              },
+            },
+          },
+          success: {
+            entry: 'sendDataBackupSuccessEvent',
+          },
+          failure: {
+            entry: 'sendDataBackupFailureEvent',
+          },
+        },
+        on: {
+          FETCH_DATA: {
+            target: '.checkStorageAvailability',
+          },
+          OK: {
+            target: '.idle',
+          },
+          DISMISS: {
+            target: 'init',
+          },
+        },
+      },
+    },
+  },
+  {
+    actions: {
+      setDataFromStorage: model.assign({
+        dataFromStorage: (_context, event) => {
+          return event.response;
+        },
+      }),
+
+      setFileName: model.assign({
+        fileName: (_context, event) => {
+          return event.filename;
+        },
+      }),
+
+      fetchAllDataFromDB: send(StoreEvents.EXPORT(), {
+        to: context => context.serviceRefs.store,
+      }),
+
+      sendDataBackupStartEvent: () => {
+        sendStartEvent(
+          getStartEventData(TelemetryConstants.FlowType.dataBackup),
+        );
+        sendImpressionEvent(
+          getImpressionEventData(
+            TelemetryConstants.FlowType.dataBackup,
+            TelemetryConstants.Screens.dataBackupScreen,
+          ),
+        );
+      },
+
+      sendDataBackupSuccessEvent: () => {
+        sendEndEvent(
+          getEndEventData(
+            TelemetryConstants.FlowType.dataBackup,
+            TelemetryConstants.EndEventStatus.success,
+          ),
+        );
+      },
+
+      sendDataBackupFailureEvent: () => {
+        sendEndEvent(
+          getEndEventData(
+            TelemetryConstants.FlowType.dataBackup,
+            TelemetryConstants.EndEventStatus.failure,
+          ),
+        );
+      },
+    },
+
+    services: {
+      checkStorageAvailability: () => async () => {
+        return Promise.resolve(
+          Storage.isMinimumLimitReached('minStorageRequired'),
+        );
+      },
```

Comment:
This logic should ideally be re-written & the server preset storage limit should be added to the estimated size of the uncompressed data + ZIP.

Reply:
How is this being evaluated? Ideally some more space beyond the filesize of 2 *(MMKV + VC) raw size should exist so that the the data can be copied and an archive can be processed accordingly

Reply:
@swatigoel do we want to implement this? This looks good and I did some exploration. We can do it.